src/main/java/nl/tudelft/pixelperfect/event/EventParameter.java

```diff
@@ -56,17 +56,67 @@ public String getKey() {
    * 
    * @return A boolean.
    */
-  public boolean isContinuous() {
+  public boolean isGeneric() {
     return (value == EventParameterValues.GENERIC);
   }
 
   /**
-   * Check whether this parameter has a discrete value.
+   * Get this parameter's number value.
    * 
-   * @return A boolean.
+   * @return A number value.
+   */
+  protected int getNumber() {
+    return number;
+  }
+
+  /**
+   * Get this parameter's specific value.
+   * 
+   * @return A specific value.
+   */
+  protected EventParameterValues getValue() {
+    return value;
+  }
+
+  /**
+   * Compares two EventParameters to see if they're the same.
+   * 
+   * @param that
+   *          The object to compare this instance to.
+   * 
+   * @return Whether or not the compared objects are the same EventParameter.
+   */
+  @Override
+  public boolean equals(Object that) {
+    if (that instanceof EventParameter) {
+      EventParameter other = (EventParameter) that;
+      if (key != other.getKey()) {
+        return false;
+      }
+      if (isGeneric() && other.isGeneric()) {
+        return number == other.getNumber();
+      } else {
+        return value.equals(other.getValue());
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Generate a Hash Code for this instance.
+   * 
+   * @return A likely unique hash code for this instance.
    */
-  public boolean isDiscrete() {
-    return !isContinuous();
+  @Override
+  public int hashCode() {
+    int result = 42;
+    int prime = 37;
```

Comment:
No they are class scoped, not application scoped. Each hashCode method may use different numbers for these purposes.